### PR TITLE
chore(core): PHP 8.2 compatibility

### PR DESCRIPTION
**Pull request by jeabakker**
ref #14237

for translation cleanup

- [x] cleanup plugin unused keys
- [x] check plugin translations for keys that don't exist in English
- [x] check core for unused keys
- [x] check core translations for keys that don't exist in English
- [x] remove English translations from translation files (due to wrong import from Transifex)

### CODE REVIEW

File: engine/classes/Elgg/Exceptions/PluginException.php [hunk 1]
```diff
@@ -11,6 +11,11 @@
  */
 class PluginException extends Exception {
 
+	/**
+	 * @var \ElggPlugin|null
```
Comment (jdalsem) [inline engine/classes/Elgg/Exceptions/PluginException.php]: no need

File: engine/tests/classes/Elgg/Helpers/Collections/TestItem.php [hunk 2]
```diff
@@ -9,13 +9,24 @@
  */
 class TestItem implements CollectionItemInterface {
 	
+	/**
+	 * @var string|int
+	 */
+	protected $id;
```
Comment (jdalsem) [inline engine/tests/classes/Elgg/Helpers/Collections/TestItem.php:15]: type hint attributes and remove docs

File: engine/tests/phpunit/integration/Elgg/Database/Clauses/AccessWhereClauseIntegrationTest.php [hunk 3]
```diff
@@ -9,6 +9,11 @@
  */
 class AccessWhereClauseIntegrationTest extends IntegrationTestCase {
 
+	/**
+	 * @var \ElggUser
+	 */
+	protected $user;
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Database/Clauses/AccessWhereClauseIntegrationTest.php:15]: type hint attribute (nullable)

File: engine/tests/phpunit/integration/Elgg/Integration/ElggCoreMetastringsTest.php [hunk 4]
```diff
@@ -23,6 +23,11 @@ class ElggCoreMetastringsTest extends IntegrationTestCase {
 		'metadata' => 'metadata',
 		'annotation' => 'annotations',
 	];
+	
+	/**
+	 * @var \ElggObject
+	 */
+	protected $object;
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Integration/ElggCoreMetastringsTest.php:30]: type hint

Comment (jdalsem) [reply engine/tests/phpunit/integration/Elgg/Integration/ElggCoreMetastringsTest.php:30]: and nullable

File: engine/tests/phpunit/integration/Elgg/Navigation/MenuRenderingIntegrationTest.php [hunk 5]
```diff
@@ -10,7 +10,22 @@
  * @group Navigation
  */
 class MenuRenderingIntegrationTest extends IntegrationTestCase {
-
+	
+	/**
+	 * @var \ElggUser
+	 */
+	protected $user;
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Navigation/MenuRenderingIntegrationTest.php:17]: atrtibutes type hints and nullables

File: engine/tests/phpunit/integration/Elgg/Notifications/NotificationsServiceIntegrationTestCase.php [hunk 6]
```diff
@@ -31,6 +31,16 @@ abstract class NotificationsServiceIntegrationTestCase extends IntegrationTestCa
 	 */
 	protected $session;
 
+	/**
+	 * @var \Elgg\SessionManagerService
+	 */
+	protected $session_manager;
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Notifications/NotificationsServiceIntegrationTestCase.php:37]: typehinting

File: engine/tests/phpunit/integration/Elgg/Views/DateOutputTest.php [hunk 7]
```diff
@@ -11,6 +11,16 @@
  */
 class DateOutputTest extends ViewRenderingTestCase {
 
+	/**
+	 * @var DateTime
+	 */
+	protected $date;
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Views/DateOutputTest.php:17]: typehinting

File: engine/tests/phpunit/unit/Elgg/Security/UrlSignerUnitTest.php [hunk 8]
```diff
@@ -14,7 +14,12 @@ class SignedUrlUnitTest extends \Elgg\UnitTestCase {
 	/**
 	 * @var UrlSigner
 	 */
-	private $service;
+	protected $service;
```
Comment (jdalsem) [inline engine/tests/phpunit/unit/Elgg/Security/UrlSignerUnitTest.php:17]: typehinting

**Review by jdalsem — CHANGES_REQUESTED** [review]

**Comment by jeabakker** [comment]
maybe/probably I haven't found all English translations in other translations, but I found a lot

**Review by jdalsem — APPROVED** [review]

**Review by jdalsem — APPROVED** [review]

**Pull request merged**